### ISSUE TITLE
Mapping attribute in class DynamicTemplate does not work as expected

### ISSUE THREAD

**Issue by matthewyang0807** [elastic/elasticsearch-net#7234]
**Elastic.Clients.Elasticsearch version**: 8.0.5

**Elasticsearch version**: 8.0

**.NET runtime version**:  6.0

**Operating system version**: Win11

**Description of the problem including expected versus actual behavior**:

**Steps to reproduce**:
 1.      public static Union<IDictionary<string, DynamicTemplate>?, ICollection<IDictionary<string, DynamicTemplate>>?> 
          GetDynamicTemplates()
        {
            IDictionary<string, DynamicTemplate> templatesDic1 = new Dictionary<string, DynamicTemplate>();
            ICollection<IDictionary<string, DynamicTemplate>> templatesDicList = new List<IDictionary<string, DynamicTemplate>>();

            var template1 = new DynamicTemplate
            {
                PathMatch = "testPathMatch",
                Mapping = new Properties
                {
                    **//The problem is here, I have tried many different types of property to add here, but will always lead the dynamic 
                   //template creation failed**
                    { "testType", new KeywordProperty() }
                }
            };

            templatesDic1.Add("testTemplateName", template1);
            templatesDicList.Add(templatesDic1);
            return new Union<IDictionary<string, DynamicTemplate>?, ICollection<IDictionary<string, DynamicTemplate>>?> 
             (templatesDicList);
        }
 3.       CreateIndexRequest cr = new CreateIndexRequest("matthew-es8-typeahead1");
           cr.Mappings.DynamicTemplates = GetDynamicTemplates();
           var indexCreateResponse = es8client.Indices.CreateAsync(cr);
           Console.WriteLine(indexCreateResponse.Result.Acknowledged);
           Console.WriteLine(indexCreateResponse.Result.ApiCallDetails.HttpStatusCode);
           Console.WriteLine(indexCreateResponse.Result.ApiCallDetails.OriginalException);
 4.      It will report the error 
![image](https://user-images.githubusercontent.com/117877546/218914988-f7873654-62e3-45b7-8e26-3e81efe3bf30.png)
 
I tried many times there is no property can be put here to make dynamic template created succsesfully.
I notice that old version in NEST6 the Mapping property type is IProperty not Properties, so can you provide me a correct way to generate the DynamicTemple? 

**Expected behavior**

I want to see my index "matthew-es8-typeahead1" with correct dynamic template in mapping part
"dynamic_templates": [ 
"testTemplateName": {
        "path_match": "testPathMatch",
        "mapping": {
          "type": "keyword"}
}]

**Provide `DebugInformation` (if relevant)**:
Elastic.Transport.TransportException: Request failed to execute. Call: Status code 400 from: PUT /matthew-es8-typeahead1. ServerError: Type: mapper_parsing_exception Reason: "Failed to parse mapping: dynamic template [testTemplateName] has invalid content [{"path_match":"testPathMatch","mapping":{"testType":{"type":"keyword"}}}], attempted to validate it with the following match_mapping_type: [object, string, long, double, boolean, date, binary]" CausedBy: "Type: illegal_argument_exception Reason: "dynamic template [testTemplateName] has invalid content [{"path_match":"testPathMatch","mapping":{"testType":{"type":"keyword"}}}], attempted to validate it with the following match_mapping_type: [object, string, long, double, boolean, date, binary]" CausedBy: "Type: mapper_parsing_exception Reason: "unknown parameter [testType] on mapper [__dynamic__testTemplateName] of type [null]"""
400

**Comment by stevejgordon** [comment]
This is related to #7221. I expect some changes to the types when we fix the generated code, which may resolve this as a by-product. If not, we can investigate this after a PR is merged, fixing the first issue.

**Comment by stevejgordon** [comment]
@matthewyang0807 A fixed has been merged which will ship in the next patch release. Once that's available you can achieve your objective with the code along the lines of:

```c#
var myTemplate = new DynamicTemplate
{
   PathMatch = "testPathMatch",
   Mapping = new KeywordProperty()
};

var createIndexRequest = new CreateIndexRequest("testing")
{
   Mappings = new TypeMapping
   {
      DynamicTemplates = new[]
      {
         new Dictionary<string, DynamicTemplate>
         {
            { "testTemplateName", myTemplate }
         }
      }
   }
};

var indexCreateResponse = await client.Indices.CreateAsync(createIndexRequest);
```

**Comment by matthewyang0807** [comment]
@stevejgordon 
Is this fix included in version 8.0.6? I run this sample code in v8.0.6 will still get an dynamic template parsing error...or do I miss something?

Error Below:

Elastic.Transport.TransportException: Request failed to execute. Call: Status code 400 from: PUT /matthew-fix-dynamictemplate-test. ServerError: Type: mapper_parsing_exception Reason: "Failed to parse mapping: Dynamic template syntax error. An array of named objects is expected." CausedBy: "Type: mapper_parsing_exception Reason: "Dynamic template syntax error. An array of named objects is expected.""

**Comment by stevejgordon** [comment]
@matthewyang0807 The original exception was caused by a malformed request. This new error is subtly different but also due to the request format not being excepted by the server. In this case, it appears to be an issue in the ES specification, which I'll review and fix before we code-gen this request again. I'll reopen to track.
